prov/util/src/util_mr.c

```diff
@@ -152,14 +154,25 @@ static int compare_mr_keys(void *key1, void *key2)
 }
 
 
+/*
+ * If a provider or app whose version is < 1.5, calls this function and passes
+ * FI_MR_UNSPEC as mode, it would be treated as FI_MR_SCALABLE.
+ */
 int ofi_mr_map_init(const struct fi_provider *prov, enum fi_mr_mode mode,
 		    struct ofi_mr_map *map)
 {
 	map->rbtree = rbtNew(compare_mr_keys);
 	if (!map->rbtree)
 		return -FI_ENOMEM;
 
-	map->mode = mode;
+	switch (mode) {
+	case FI_MR_BASIC:
+		map->mode = OFI_MR_BASIC_MAP;
+	case FI_MR_SCALABLE:
+		map->mode = 0;
+	default:
+		map->mode = mode;
+	}
```

Comment:
>There are no break statements in the switch cases

Oops. Will fix that. 

I think the function needs to be updated to take flags. The default case would then be okay as it is.